include/usearch/index.hpp

```diff
@@ -2929,6 +3060,52 @@ class index_gt {
         return result;
     }
 
+    /**
+     *  @brief  Merge an index to this index.
+     *
+     *  @param[in] index The index that has nodes to be merged.
+     *  @param[in] get_value Callable object getting a value from `index`.
+     *  @param[in] metric Callable object measuring distance between value and present objects.
+     *  @param[in] metric Callable object measuring distance between value and present objects.
+     *  @param[in] config Configuration options for this specific operation.
+     *  @param[in] callback On-success callback, executed while the `member_ref_t` is still under lock.
+     */
+    template <                                               //
+        typename metric_at,                                  //
+        typename get_value_at,                               //
+        typename merge_callback_at = dummy_merge_callback_t, //
+        typename prefetch_at = dummy_prefetch_t              //
+        >
+    add_result_t merge(                                                      //
+        index_gt const& index, get_value_at&& get_value, metric_at&& metric, //
+        index_update_config_t update_config = {}, merge_callback_at&& callback = merge_callback_at{},
+        prefetch_at&& prefetch = prefetch_at{}) noexcept {
+        // Reserve spaces.
+        std::size_t new_members = size() + index.size();
+        index_limits_t limits;
+        limits.members = new_members;
+        if (!reserve(limits)) {
+            add_result_t result;
+            return result.failed("Out of memory");
+        }
+
+        // Add all values in `index` to this index.
+        add_result_t merge_result;
+        for (const auto& member : index) {
+            auto& value = get_value(member);
```

Comment:
If we want to filter the target values as suggested in https://github.com/unum-cloud/usearch/issues/84#issuecomment-2300410243 , we can do it here.